src/thunderbots/software/ai/navigator/path_planning_navigator/path_planning_navigator.cpp

```diff
@@ -255,3 +302,30 @@ void PathPlanningNavigator::drawObstacle(const Obstacle &obstacle,
             circleToPolygon(*obstacle.getBoundaryCircle(), 12), 0.025, color);
     }
 }
+
+double PathPlanningNavigator::getCloseToEnemyObstacleFactor(Point &p)
+{
+    double closest_dist = DBL_MAX;
+    for (auto &robot : world.enemyTeam().getAllRobots())
+    {
+        Obstacle o = Obstacle::createRobotObstacleWithScalingParams(
+            robot,
+            Util::DynamicParameters::Navigator::robot_obstacle_inflation_factor.value(),
+            Util::DynamicParameters::Navigator::velocity_obstacle_inflation_factor
+                .value());
+        double current_dist = dist(p, (*o.getBoundaryPolygon()));
+        if (current_dist < closest_dist)
+        {
+            closest_dist = current_dist;
+        }
+    }
+
+    if (closest_dist > 2)
+    {
+        return 1;
+    }
+    else
+    {
+        return closest_dist / 2;
+    }
```

Comment:
I'm a bit confused about this logic. Are we trying to avoid enemies by a larger distance if we are farther away?

Reply:
it's basically a mapping of 0 to 1 onto 0 to 2 to get a factor for how fast the robot should go, so if the robot is farther away from obstacles then it can go faster and vice versa